### ISSUE TITLE
Focus Issues in chrome during tiling

### ISSUE THREAD

**Issue by piyushk** [qtile/qtile#1824]
# Issue description

When chrome is not floating in a tiling layout (MonadTall/Tile), I can't click on various interface boxes. Here are 2 examples:
- Create a bookmark by clicking star. The Bookmark Added/Edit Bookmark popup appears. You cannot click on any option in this popup. All clicks are captured by the underlying chrome window instead.
- If I type in a query in the chrome bar, I can't click on any of the autocomplete suggestions. I can hit the down arrow to select them though

If chrome is floating and not tiled, then these problems do not occur.

# Qtile version

Tested on latest master: https://github.com/qtile/qtile/commit/e87b32b7c5a8652f077f09441705e20286f2d6f7
Present in 0.15.1

I noticed this problem a long time ago, so it's been around for a few versions.

# Configuration

Same as https://github.com/qtile/qtile/blob/master/libqtile/resources/default_config.py, except with MonadTall.
```
layouts = [
    # layout.Max(),
    # layout.Stack(num_stacks=2),
    # Try more layouts by unleashing below layouts.
    # layout.Bsp(),
    # layout.Columns(),
    # layout.Matrix(),
    layout.MonadTall(),
    # layout.MonadWide(),
    # layout.RatioTile(),
    # layout.Tile(),
    # layout.TreeTab(),
    # layout.VerticalTile(),
    # layout.Zoomy(),
]
```

**Comment by ramnes** [comment]
Yes, I can confirm, thank you for writing it down, super annoying. Can you also confirm that it does not happen with the Max layout? @tych0, that's what I was telling you about during last FOSDEM.

**Comment by piyushk** [comment]
@ramnes I can confirm it doesn't happen with the max layout.

**Comment by tych0** [comment]
Yep, I can reproduce this on other layouts. Never noticed because I use the max layout for my browser :)

**Comment by tych0** [comment]
I spent some time thinking about this today. It turns out that the bookmarks window is not actually an X window, so qtile isn't mis-focusing or anything.

That it only works in Max suggests that there's something about the offset calculation in chrome that's weird. One thing of note is that it has:

```_NET_WM_STATE(ATOM) = _NET_WM_STATE_MAXIMIZED_VERT, _NET_WM_STATE_MAXIMIZED_HORZ```

which suggests that it thinks it's fullscreened independently of axis. Qtile doesn't support these settings at all, so it just ignores them. This suggests to me again that maybe if it thinks it's fullscreend, something about the click math is off. I tried just removing these atoms (hoping that chrome would be smart enough to fix itself), but it didn't seem to work: https://github.com/tych0/qtile/commit/fceb46311d54251e3d4c3a7e42f75b3e628e8034

Anyway, that's all for now :)

**Comment by mikelward** [comment]
I think I'm seeing the same issue. When I click on an password or credit card autocomplete dropdown in Chrome, the dropdown is dismissed, but the field is not filled out.

**Comment by mikelward** [comment]
Seems to happen going back to at least v0.10.0

**Comment by mikelward** [comment]
@tych0 If I click Chrome's own maximize/restore button, `_NET_WM_STATE(ATOM)` changes to empty, but I can still repro.

Another thing I notice is that the click seems to be passed thru to whatever was under the autocomplete overlay. e.g. on my bank's login page, the "Remember me" checkbox is under the autocomplete dropdown. If I click the autocomplete suggestion in just the right place, not only is the overlay dismissed, but the "Remember me" checkbox state is toggled.

**Comment by tramhao** [comment]
This problem is driving me crazy. I'm using brave browser but has the same problem as chrome. Normally I use tiling layout for browser because I like the gaps. Now I have to use max layout and no gaps. Do you guys have some updates on this issue? Thanks so much.

**Comment by Graeme22** [comment]
+1 for this on brave

**Comment by piyushk** [comment]
I'm afraid that Chrome `89.0.4389.90` is pretty much unusable with qtile regardless of tiling mode. In both max and monadtall, a number of custom menus on various websites are unclickable. Example: the volume contron in netflix can no longer be controlled.

**Comment by neworld** [comment]
I found middle mouse button sometimes works just fine or at least works by
opening the page in the tab. I am curious, what magic is chrome using?

Best wishes,
Andrius Semionovas

**Comment by m-col** [comment]
Does https://github.com/qtile/qtile/pull/2506 fix it?

**Comment by piyushk** [comment]
@m-col Thanks for the PR

I tried to switch qtile to the branch in that PR and encountered some problems but it won't start. same for the current qtile master. I assume there's something about my config it doesn't like, but looking at the qtile log with or without xtrace did not help.

I'll test with the default config soon and get back to you.

**Comment by m-col** [comment]
If you are going from the last release to master then you may have to run `qtile migrate` if something has changed in the way it is configured. Might be worth trying that.

On 210603 10:27, Piyush Khandelwal wrote:
***@***.*** Thanks for the PR

**Comment by piyushk** [comment]
@m-col After some config wrangling, I was able to test both the latest stable release (0.17.0) and your PR branch. I'm afraid the 2 problems in this issue report are still not solved.

Since there is a minor change since the initial issue description, I'll reiterate the problems that currently exist in chrome:
- In a tiling layout, you can't click on some chrome interfaces. See the bookmark example in my initial issue description. **Different from my initial report making a window floating in a tiling layout no longer fixes the issue.**
- In more recent versions of chrome, various custom menus are not left-clickable. I believe this is true regardless of layout. The volume bar in Netflix is one such example. Middle-click works in these settings.

**Comment by ramnes** [comment]
I just tried it and everything you mentioned in the first post got fixed for me. Are you sure you were using @m-col's branch, @piyushk? Also it's merged in master now so it may be easier to try. I will close the issue based on my own tests but feel free to comment and will try to understand what's happening if you still see this.

**Comment by piyushk** [comment]
I can confirm that the issue persists for me with `mondatall` using the latest master with the default config (w/ `monadtall` uncommented) as well as my usual config.
```
piyushk@bird-person:~/code$ qtile --version
0.17.1.dev315+g67f97604
```
[config.py.txt](https://github.com/qtile/qtile/files/6610698/config.py.txt) (default config with `monadtall` enabled)
[config.py.piyushk.txt](https://github.com/qtile/qtile/files/6610709/config.py.piyushk.txt) (my usual config as is)

**Comment by piyushk** [comment]
@ramnes @m-col chrome seems to have made adding bookmarks a two step process since the original report. Can you try the following? Click Star + Add/Edit Bookmark (works) + More/Remove/Folder Dropdown (does not work, click captured by underlying chrome window). 

I positioned the github "New Issue" button below the Name entry in the "Add/Edit Bookmark dialog", and clicking on the "Name" entry navigates to a new issue page.

I've confirmed that it works in max as expected.

**Comment by ramnes** [comment]
Looks like I didn't test properly, shame on me. :) 

I think I tried it with the Max layout that didn't work before #2506 either, IIRC.

By the way, a simpler test is just to click on an autocompletion suggestion in the navigation bar.

**Comment by brkastner** [comment]
> @m-col After some config wrangling, I was able to test both the latest stable release (0.17.0) and your PR branch. I'm afraid the 2 problems in this issue report are still not solved.
> 
> Since there is a minor change since the initial issue description, I'll reiterate the problems that currently exist in chrome:
> 
> * In more recent versions of chrome, various custom menus are not left-clickable. I believe this is true regardless of layout. The volume bar in Netflix is one such example. Middle-click works in these settings.

I'm seeing this issue with custom menus using the stable version of Chrome (v91.0.4472.101-1) but the beta (92.0.4515.40-1) and dev (93.0.4535.3-1) versions from the AUR are working correctly! Bookmarks and other chrome interfaces are still not working though.

**Comment by dodopod** [comment]
Works for me, too, after upgrading to qtile v0.18.1.

**Comment by m-col** [comment]
Awesome, thank you for reporting! I believe https://github.com/stonewell/qtile/commit/541a3ad1b189675218c5b179ef741b54076a1705 will have fixed it.

If anybody continues to have issues feel free to post and we can re-open.

**Comment by angrybacon** [comment]
Thanks for the `migrate` command, wasn't aware of it! Fortunately, the issue was much more dumb: I had `qtile` in my ignored package so I wasn't pulling the fix that happened from that thread. I did try 18 (18.0?) on another machine (hence my previous post) but the issue seems to be fixed as of 18.1 :raised_hands: 

> What troubles are you having with updating?

I was starting Qtile with `exec qtile` instead of `qtile start` in my `~/.xinitrc` too

Thanks again for the follow up :pray:
